diff --git a/configure.ac b/configure.ac
index 6ef750f..a31ea1a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -508,7 +508,9 @@ AM_CONDITIONAL(BUILDDOCS, test "x$BUILDDOCS" = xyes)
 
 dnl Only build sgml docs when linuxdoc is available and
 dnl def.ents has been installed
-XORG_CHECK_LINUXDOC
+dnl XORG_CHECK_LINUXDOC
+AM_CONDITIONAL(BUILD_LINUXDOC, false)
+AM_CONDITIONAL(BUILD_PDFDOC, false)
 
 dnl Handle installing libxf86config
 AM_CONDITIONAL(INSTALL_LIBXF86CONFIG, [test "x$INSTALL_LIBXF86CONFIG" = xyes])