From f1d003ce1de302a2a72447b175251dc845210d01 Mon Sep 17 00:00:00 2001
From: Andreas Zuber <zuber@puzzle.ch>
Date: Sat, 21 Apr 2012 13:59:45 +0200
Subject: [PATCH] TERA FIX 001

---
 dlls/user32/winstation.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/dlls/user32/winstation.c b/dlls/user32/winstation.c
index 8126b31..b3a8a91 100644
--- a/dlls/user32/winstation.c
+++ b/dlls/user32/winstation.c
@@ -462,9 +462,7 @@ BOOL WINAPI EnumDesktopsW( HWINSTA winsta, DESKTOPENUMPROCW func, LPARAM lparam
  */
 HDESK WINAPI OpenInputDesktop( DWORD flags, BOOL inherit, ACCESS_MASK access )
 {
-    FIXME( "(%x,%i,%x): stub\n", flags, inherit, access );
-    SetLastError( ERROR_CALL_NOT_IMPLEMENTED );
-    return 0;
+    return CreateDesktopA("Desktop", 0, 0, flags, access, 0); 
 }
 
 
-- 
1.7.3.4

