From a9c7dbae6622c9bcb757448e8f77419a1d372557 Mon Sep 17 00:00:00 2001
From: Andreas Zuber <zuber@puzzle.ch>
Date: Mon, 24 Jul 2017 23:55:21 +0200
Subject: [PATCH] [PATCH] Append /pyfa to import path

---
 pyfa.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pyfa.py b/pyfa.py
index 4faa1edd..9cfa5e23 100755
--- a/pyfa.py
+++ b/pyfa.py
@@ -23,6 +23,7 @@ import os
 import platform
 import re
 import sys
+sys.path.append(os.path.join("%%SITEDIR%%", "pyfa"))
 import traceback
 from optparse import AmbiguousOptionError, BadOptionError, OptionParser
 
-- 
2.13.3

